### PR TITLE
Add support for libavif >= 0.6.0

### PR DESCRIPTION
**Pull request by cryptomilk**
Hi,

libavif 0.6.0 has a new API for handling RGB which is not compatible with libavif < 0.6.0. I've added support for the new API and the patches for adding support can be cherry-picked for darktable 0.6.0.

For darktable 3.1 we will require libavif >= 0.6.0 and remove the old code.

### CODE REVIEW

File: src/imageio/format/avif.c [hunk 1]
```diff
@@ -228,56 +231,54 @@ int write_image(struct dt_imageio_module_data_t *data,
...
   case 8: {
 #ifdef _OPENMP
 #pragma omp parallel for simd default(none) \
-  dt_omp_firstprivate(in_data, width, height, R_plane8, G_plane8, B_plane8, \
-                      R_rowbytes, G_rowbytes, B_rowbytes, max_channel_f) \
+  dt_omp_firstprivate(in_data, width, height, out, rowbytes, max_channel_f) \
   schedule(simd:static) \
   collapse(2)
 #endif
     for (size_t y = 0; y < height; y++) {
```
Comment (TurboGit) [inline src/imageio/format/avif.c]: { should be on next line.

likewise for next {

**Comment by TurboGit** [comment]
> I've added support for the new API and the patches for adding support can be cherry-picked for darktable 0.6.0.

darktable 3.0.2 ?

> For darktable 3.1 we will require libavif >= 0.6.0 and remove the old code.

You mean for 3.2 here, right?

So please can you clarify ?

**Comment by cryptomilk** [comment]
I thought you picked it for 3.0.1. Doesn't seem to be the case.

**Comment by ptilopteri** [comment]
obs openSUSE Tumbleweed build 960 is failing

**Comment by ptilopteri** [comment]
> obs openSUSE Tumbleweed build 960 is failng

only libavif0-0.5.5-1.2.x86_64 is available, no 0.6...

**Comment by cryptomilk** [comment]
You should take a look at the build log:

```
[   40s] libavif2-0.6.2-1.1                    ########################################
```

it is clearly installing libavif 0.6.2 ...

**Comment by ptilopteri** [comment]
> You should take a look at the build log:
> 
> ```
> [   40s] libavif2-0.6.2-1.1                    ########################################
> ```
> 
> it is clearly installing libavif 0.6.2 ...

ok, but it *is* failing and avifImage appears to be the problem,

[  211s]
/home/abuild/rpmbuild/BUILD/darktable-3.1.0~git963.4f0575b82/src/common/imag
eio_avif.c:
/In function 'dt_imageio_open_avif':
[  211s]
/home/abuild/rpmbuild/BUILD/darktable-3.1.0~git963.4f0575b82/src/common/imag
eio_avif.c:160:12:
/error: too few arguments to function 'avifImageYUVToRGB'
[  211s]   160 |   result = avifImageYUVToRGB(avif);
[  211s]       |            ^~~~~~~~~~~~~~~~~
[  211s] In file included from
/home/abuild/rpmbuild/BUILD/darktable-3.1.0~git963.4f0575b82/src/common/imag
eio_avif.c:22:

among many other lines

**Comment by cryptomilk** [comment]
Yes, libavif 0.6 has a differnt API that's why we have this pull request, to support the new API.

**Comment by ptilopteri** [comment]
@cryptomilk 
but this commit has broken the obs build for opensuse Tumbleweed which darix claims you support

the last successful build was git822.a06f8db58

**Comment by cryptomilk** [comment]
@ptilopteri Which commit broke it exactly?

**Comment by ptilopteri** [comment]
@cryptomilk   I have no way to tell, except that it has failed since git822.a06f8db58

here is the failure log:
  https://build.opensuse.org/package/live_build_log/graphics:darktable:master/darktable/openSUSE_Tumbleweed/x86_64

maybe ask darix for help ???

**Comment by cryptomilk** [comment]
The log shows that it tires to build against libavif 0.6.2 and darktable doesn't have support for that API version yet. This pull request addresses it. How often do I have to repeat that?

**Comment by ptilopteri** [comment]
it is good that you have addressed it, but that has not solved the obs darktable:master build for Tumbleweed.  I presented you the failure(s) and it is still failing.  I do not know if simply presenting a pull request is a solution.  It still is not building.

**Comment by parafin** [comment]
PR has to be merged first to take effect;) basically obs build fails, because wrong version of dependency (libavif) is used.

**Comment by ptilopteri** [comment]
OBS Build is still failing but I do not know why.

failure text: https://build.opensuse.org/package/live_build_log/graphics:darktable:master/darktable/openSUSE_Tumbleweed/x86_64

**Comment by ptilopteri** [comment]
more about failure:

nothing provides libicuuc.so.suse65.1()(64bit) needed by libraptor2-0, nothing provides libicuuc.so.suse65.1()(64bit) needed by libharfbuzz-icu0, nothing provides libicuuc.so.suse65.1()(64bit) needed by libQt5Core5, nothing provides libicui18n.so.suse65.1()(64bit) needed by libQt5Core5

for obs Tumbleweed build failure

**Comment by parafin** [comment]
How is this darktable's problem? I think you're posting at the wrong place.

**Comment by ptilopteri** [comment]
@parafin    darix said the dt dev's were maintaining the OBS.  And was somewhat emphatic that he was not.

**Comment by cryptomilk** [comment]
If you have issues with packages for tumbleweed, the right place to report them is https://bugzilla.opensuse.org/

**Pull request merged**